src/nimlsp.nim

```diff
@@ -371,7 +373,36 @@ while true:
                   )
                 ).JsonNode
               message.respond response
-
+        of "textDocument/documentSymbol":
+          message.textDocumentRequest(DocumentSymbolParams, symbolRequest):
+            debugEcho "Running equivalent of: outline ", fileuri[7..^1], ";", filestash
+            let syms = getNimsuggest(fileuri).outline(fileuri[7..^1], dirtyfile = filestash)
+            debugEcho "Found outlines: ",
+              syms[0..(if syms.len > 10: 10 else: syms.high)],
+              (if syms.len > 10: " and " & $(syms.len-10) & " more" else: "")
+            if syms.len == 0:
+              message.respond newJNull()
+            else:
+              var response = newJarray()
+              for sym in syms.sortedByIt((it.line,it.column,it.quality)):
+                debugEcho $sym
```

Comment:
This should be removed, the above `debugEcho` tries to limit the amount of entries to 10 in order to not flood the log, but then this prints all symbols, one per line.